sched/definitions.go

```diff
@@ -17,3 +17,33 @@ type Task struct {
 	Command    []string // Argv to run
 	SnapshotId string
 }
+
+type JobState int
+
+const (
+	// An unambiguous 0-value.
+	UNKNOWN JobState = iota
+	// Waiting to run.
+	PENDING
+	// Running
+	RUNNING
+	// Done
+	DONE
+)
+
+type JobStatus struct {
```

Comment:
I think people should use the factory methods to create, but want to just read the data. I wish there were immutable types, so that we could ensure clients don't modify the fields, but, well, go.